
More info on this patch:

  http://www.kaffe.org/pipermail/kaffe/2003-June/042426.html

This should be merged upstream to libtool.

 
--- libltdl/ltdl.c.jim	2003-06-06 16:55:53.000000000 -0700
+++ libltdl/ltdl.c	2003-06-06 16:56:37.000000000 -0700
@@ -237,7 +237,10 @@
 #else
 
 #define LT_DLMALLOC(tp, n)	((tp *) lt_dlmalloc ((n) * sizeof(tp)))
+/*
 #define LT_DLREALLOC(tp, p, n)	((tp *) rpl_realloc ((p), (n) * sizeof(tp)))
+*/
+#define LT_DLREALLOC(tp, p, n)	((tp *) lt_dlrealloc ((p), (n) * sizeof(tp)))
 #define LT_DLFREE(p)						\
 	LT_STMT_START { if (p) (p) = (lt_dlfree (p), (lt_ptr) 0); } LT_STMT_END
 
